### PR TITLE
Prepare for Crystal 0.24

### PR DESCRIPTION
**Pull request by lipanski**
Fixes for the breaking changes to be introduced by the [0.24 pre-release](https://github.com/crystal-lang/crystal/releases/tag/0.24.0).

Let me know what you think.

Tests are passing locally, but it will probably take some time for Travis to pick up the new version, so I guess we can't release it just yet.

screenshot-2017-11-19_00-40-11

### CODE REVIEW

**Comment by lipanski** [comment]
I can't really set the fork to point to a `crystal-0.24` branch unless it exists, so could you please create & push it?

File: src/mysql/types.cr [hunk 1]
```diff
@@ -191,15 +191,15 @@ abstract struct MySql::Type
...
+      return ::Time::Span.new(nanoseconds: 0) if pkt < 1
       negative = packet.read_byte!.to_i32
       days = packet.read_fixed_int(4)
       hour = packet.read_byte!.to_i32
       minute = packet.read_byte!.to_i32
       second = packet.read_byte!.to_i32
-      ms = pkt > 8 ? (packet.read_int.to_i32 / 1000) : nil
-      time = ms ? ::Time::Span.new(days, hour, minute, second, ms) : ::Time::Span.new(days, hour, minute, second)
-      negative > 0 ? (::Time::Span.new(0) - time) : time
+      ns = pkt > 8 ? (packet.read_int.to_i32 * 1000) : nil
```
Comment (lipanski) [inline src/mysql/types.cr]: I'm assuming `packet.read_int.to_i32` delivers microseconds, correct?

**Comment by lipanski** [comment]
@bcardiff would you care to have a look?

**Comment by bcardiff** [comment]
Thanks @lipanski the PR was great! Now I was trying to add some specs to ensure microseconds are stored... but I am facing some issues. Even so, the PR keeps the current functionally just fine.

I changed the base branch so this don't arrive master yet. Contribs to crystal v0.24 can keep going into that branch.

**Comment by lipanski** [comment]
@bcardiff thanks! :cake:

**Comment by bcardiff** [comment]
"some issues" revealed https://github.com/crystal-lang/crystal/pull/5317 😅

**Comment by lipanski** [comment]
@bcardiff better now than later :+1:

**Pull request merged**